app/views/core_induction_programmes/modules/show.html.erb

```diff
@@ -35,7 +35,13 @@
     <ul class="app-task-list__items">
       <% if @course_lessons_with_progress.each do |lesson| %>
         <li class="app-task-list__item">
-          <h2 class="govuk-heading-m"><%= lesson.title %></h2>
+          <h2 class="govuk-heading-m">
+            <% if !lesson.course_lesson_parts.any? && current_user.admin? %>
+              <%= govuk_link_to lesson.title, lesson_path(lesson) %>
+            <% else %>
+              <%= lesson.title %>
+            <% end %>
```

Comment:
I think that's an acceptable price to pay for less complexity in code - since admin views are essentially our way of managing and improving content for time being.